frappe/core/doctype/version/version.py

```diff
@@ -147,6 +147,20 @@ def get_diff(old, new, for_child=False, compare_cancelled=False):
 				new_value = new.get_formatted(df.fieldname) if new_value else new_value
 
 			if old_value != new_value:
+				doctype = new.doctype or old.doctype
+				if doctype:
+					meta = frappe.get_meta(doctype)
+
+					if (field_meta := meta.get_field(df.fieldname)) and field_meta.fieldtype == "Link":
+						link_meta = frappe.get_meta(field_meta.options)
+
+						# Show title field value if field is Link and show_title_field_in_link is True
+						if link_meta.show_title_field_in_link:
+							title_field = link_meta.get_title_field()
+							old_title_val = frappe.get_value(field_meta.options, old_value, title_field)
+							new_title_val = frappe.get_value(field_meta.options, new_value, title_field)
```

Comment:
* I think it would be prudent to ensure that a title_field is set before making additional SELECT queries
* Title fetching can be done in one query

```suggestion
						if link_meta.show_title_field_in_link and ((title_field := link_meta.get_title_field()) != "name"):
							old_title_val, new_title_val = frappe.db.get_values(field_meta.options, {"name": ("in", (old_value, new_value))}, title_field, pluck=True)
```